worker/master/master.go

```diff
@@ -181,7 +201,27 @@ func (m *Controller) Start() error {
 	}
 	// Name of config map with leader election lock
 	lockName := "rainbond-appruntime-worker-leader"
-	go leader.RunAsLeader(m.ctx, m.conf.KubeClient, m.conf.LeaderElectionNamespace, m.conf.LeaderElectionIdentity, lockName, start, func() {})
+
+	// Become leader again on stop leading.
+	leaderCh := make(chan struct{}, 1)
+	go func() {
+		for {
+			select {
+			case <-m.ctx.Done():
+				return
+			case <-leaderCh:
+				logrus.Info("run as leader")
+				ctx, cancel := context.WithCancel(m.ctx)
+				defer cancel()
+				leader.RunAsLeader(ctx, m.kubeClient, m.conf.LeaderElectionNamespace, m.conf.LeaderElectionIdentity, lockName, start, func() {
+					leaderCh <- struct{}{}
```

Comment:
lose leader, the context in `start` function will be done?

Reply:
yes